src/components/Dashboard/TodoList.vue

```diff
@@ -403,7 +403,7 @@
                   </v-btn>
                   <v-btn
                     v-else
-                    class="btn-resume-payment"
+                    :class="{ 'btn-resume-payment': !isCancellableTodoItem(item) }"
```

Comment:
Also on line 397?

Reply:
I don't believe it's needed. That class is for a different kind of button ("Change Payment Type") that's irrelevant to uncancellable filings.

Reply:
You're sure a failed payment would want to display that other button? I think you get one button vs the other depending on the pay method that was used. But I don't know what a pay error would look like...

Reply:
AFAIK, the only pay error we would see (since this is a zero fee filing) is an error creating or linking the payment (with Pay API). I don't know if there are other pay errors for "not enough funds" or whatever. Needs to be looked into.

Reply:
You're right. I'll be adding the case to that line.

However, I'm looking at the different buttons and conditions we have (it all depends on FilingStatus), the only time we have a cancel payment is in this case. The other cases (FilingStatus.ERROR for example), we get the Retry Payment option which is fine I believe.

Reply:
Oh, I'm getting confused.

Do you know, on a pay error that we might see here, whether the filing is in Pending or Error status? There are several possible buttons (Change Payment Type, Resume Payment, Cancel Payment, Retry Payment).

Reply:
Sev, I went ahead and checked the possible buttons for every single `FilingStatus` that we have. The ones that have buttons which are applicable here in this case are:
- `FilingStatus.PENDING`:
![pending](https://github.com/bcgov/business-filings-ui/assets/122301442/74693975-5eae-49f4-bc8a-1ea2d5381374)

- `FilingStatus.ERROR`:
![error](https://github.com/bcgov/business-filings-ui/assets/122301442/9e998f9e-c54c-4fb8-83e0-1284c3ae397e)

Now, which one of the above we will see if the filing doesn't go through, I don't know yet. If the FilingStatus was `ERROR`, we would not usually see the other button. It's only when the FilingStatus is `PENDING` that we see the other chevron button to open the small menu.

Reply:
Curious: what does View Details show for the each case?

I'm thinking we'd get the error status if the payment can't be processed, and Retry should work. I don't know if we'd ever see the first one, but if your code handles both cases then we're safe and that fine by me.

Reply:
For PENDING:
![for pending details](https://github.com/bcgov/business-filings-ui/assets/122301442/15684568-5b40-4db8-83c9-828f0c143f2b)

For ERROR:
![for error details](https://github.com/bcgov/business-filings-ui/assets/122301442/b342a983-c036-4dc4-a9d0-df7e66f3bdcf)

Yup, I believe the retry should work (it happened with me a couple of days back in another filing and that worked). I guess I can fully test it when filer is merged.

Reply:
Thanks.

I'm _guessing_ that the payment record could be created and been interrupted even for this zero-fee filing, and Resume should set things back in motion.

I'm _guessing_ that an unsuccessful payment (creation?) is more likely, and Retry should restart it.

I think we should assume that both cases are possible and handle them accordingly. Can this conversation be resolved now?

Reply:
Yes, exactly. I don't see a situation/scenario in which this filing might get stuck. Currently all scenarios are handled.

Yes Sev! Unless you say otherwise, I believe we should be good.